doc/source/release-notes/Percona-Server-5.5.46-37.5.rst

```diff
@@ -0,0 +1,41 @@
+
+New Features
+============
+
+ |Percona Server| is now available for *Ubuntu* 15.10 (Wily).
+
+Bugs Fixed
+==========
+
+ Running ``PURGE CHANGED_PAGE_BITMAPS BEFORE 0`` could cause server to crash. Bug fixed :bug:`1368530`.
+
+ If user has configured server to Warn/Salvage then loop to estimate stats makes a safe exit if it encounters a corrupt page. This safe exit path failed to close (commit) and open (started) mtr which would then lead to an issue while closing transaction. Bug fixed :bug:`1426610`.
+
+ If the user had duplicate pid-file options in config files when running ``yum upgrade``, the upgrade would stop with error because it would think it found the duplicate pid while it was the same pid specified twice.  Bug fixed :bug:`1454917`.
+
+ Server would not start if :ref:`changed_page_tracking` feature was enabled and :variable:`innodb_flush_method` variable was set to ``O_DIRECT``. Bugs fixed :bug:`1500720` and :bug:`1498891`.
+
+ When :ref:`user_stats` are enabled, after executing any statement of the ``SHOW ...`` family that produces non-empty result, the total number of the sent rows is updated in the :table:`USER_STATISTICS` table. Bug fixed :bug:`1510953`.
+
+ Fixed the conflicting meta packages in *Debian* and *Ubuntu* distributions. ``percona-server-server`` and ``percona-server-client`` meta packages now point to the latest 5.6 release. Bug fixed :bug:`1292517`.
+
+ :table:`INNODB_CHANGED_PAGES` table was unavailable with non-default :variable:`innodb_data_home_dir` setting. Bug fixed :bug:`1364315`.
+
+ When :ref:`innodb_fake_changes_page` feature is enabled database records are not actually modified but the modification is assumed and accordingly the path is selected/forced. For workload involving ordering change ``UPDATE`` will be executed as series of ``DELETE`` and ``INSERT`` action. Given that fake-change mode is enabled actual delete will never happen but path will be selected assuming that it took place. This can cause server to hit some asserts in the forced path as the state was not marked. One such path that assumed record is marked as deleted when it was not is now fixed. Bug fixed :bug:`1395706`.
+
+ Due to the bug in the :ref:`innodb_fake_changes_page` logic, change in :variable:`innodb_fake_changes` value during lifetime of transaction used to affect the current active transactions too. This effectively meant some of the statements from a transaction went in dry-run format without making any change to the database and follow-up statements from the same transaction are executing with fake_change mode off thereby making changes to the database. Bug fixed :bug:`1395579`.
```

Comment:
Changing innodb_fake_changes variable value in the middle of transaction would have an immediate effect, that is, making part of transaction run with fake changes enabled and the rest with fake changes disabled, resulting in broken transaction. Fixed by making any innodb_fake_changes value changes becoming effective at the start of next transaction instead of next statement.